scripts/docker/Dockerfile

```diff
@@ -10,54 +10,54 @@ FROM golang:${VERSION} AS builder
 ARG CGO_ENABLED=0
 ARG BUILD_TAGS
 
-WORKDIR /go/src/github.com/hashicorp/vault
+WORKDIR /go/src/github.com/openbao/openbao
 COPY . .
 
 RUN make bootstrap \
-  && CGO_ENABLED=$CGO_ENABLED BUILD_TAGS="${BUILD_TAGS}" VAULT_DEV_BUILD=1 sh -c "'./scripts/build.sh'"
+  && CGO_ENABLED=$CGO_ENABLED BUILD_TAGS="${BUILD_TAGS}" OPENBAO_DEV_BUILD=1 sh -c "'./scripts/build.sh'"
 
 # Docker Image
 
 FROM alpine:3.13
 
-# Create a vault user and group first so the IDs get set the same way,
+# Create a bao user and group first so the IDs get set the same way,
 # even as the rest of this may change over time.
-RUN addgroup vault && \
-    adduser -S -G vault vault
+RUN addgroup bao && \
+    adduser -S -G bao bao
 
-# Set up certificates, our base tools, and Vault.
+# Set up certificates, our base tools, and OpenBao.
 RUN set -eux; \
     apk add --no-cache ca-certificates libcap su-exec dumb-init tzdata
 
-COPY --from=builder /go/src/github.com/lf-edge/openbao/bin/vault /bin/vault
+COPY --from=builder /go/src/github.com/lf-edge/openbao/bin/bao /bin/bao
```

Comment:
Ah, we've moved into the `openbao/openbao` namespace, so this breaks:

```
$ make docker-dev
... output elided ...
==> Results:
total 128M
-rwxr-xr-x. 1 root root 128M Jan 29 15:07 bao
--> 81f44a0917fc
[2/2] STEP 1/11: FROM alpine:3.13
[2/2] STEP 2/11: RUN addgroup bao &&     adduser -S -G bao bao
--> 1c3994831238
[2/2] STEP 3/11: RUN set -eux;     apk add --no-cache ca-certificates libcap su-exec dumb-init tzdata
+ apk add --no-cache ca-certificates libcap su-exec dumb-init tzdata
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
(1/5) Installing ca-certificates (20220614-r0)
(2/5) Installing dumb-init (1.2.5-r0)
(3/5) Installing libcap (2.46-r0)
(4/5) Installing su-exec (0.2-r1)
(5/5) Installing tzdata (2022f-r1)
Executing busybox-1.32.1-r9.trigger
Executing ca-certificates-20220614-r0.trigger
OK: 10 MiB in 19 packages
--> 2c9187cd7694
[2/2] STEP 4/11: COPY --from=builder /go/src/github.com/lf-edge/openbao/bin/bao /bin/bao
Error: building at STEP "COPY --from=builder /go/src/github.com/lf-edge/openbao/bin/bao /bin/bao": checking on sources under "/home/cipherboy/.local/share/containers/storage/overlay/7f7e7b301e82675b2bd2ea6b9dc496383f963b59484cca6cc5fac38b6c4018d6/merged": copier: stat: "/go/src/github.com/lf-edge/openbao/bin/bao": no such file or directory
make: *** [Makefile:61: docker-dev] Error 125
```

Reply:
@kvendingoldo I'm happy to pick this one up for you, but I'd appreciate it if you could do the DCO that @naphelps mentioned so we can get this merged without it. :-)